libs/core/lci_base/include/hpx/lci_base/lci_environment.hpp

```diff
@@ -74,11 +72,9 @@ namespace hpx { namespace util {
     private:
         static mutex_type mtx_;
         static bool enabled_;
-        static LCI_endpoint_t ep_;
-        static LCI_endpoint_t rt_ep_;
-        static LCI_endpoint_t h_ep_;
-        static LCI_comp_t rt_cq_r_;
-        static LCI_comp_t h_cq_r_;
+        static LCI_endpoint_t ep;
+        static LCI_comp_t scq;
+        static LCI_comp_t rcq;
         static std::thread* prg_thread_p;
```

Comment:
I'm feeling a bit uneasy here. Would it make sense to use a `std::shared_ptr` instead? Or possibly even a `std::unique_ptr`) This would guarantee that the memory is released at the end of execution.

Reply:
What object do you want me to use a smart pointer to point to? the prg_thread_p?

Reply:
Plain pointers always raise my suspicion. So yes, I'm referring to the `prg_thread_p` variable.

Reply:
Done!